nifi-commons/nifi-flow-encryptor/src/main/java/org/apache/nifi/flow/encryptor/AbstractFlowEncryptor.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.flow.encryptor;
+
+import org.apache.nifi.encrypt.PropertyEncryptor;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.regex.Pattern;
+
+public abstract class AbstractFlowEncryptor implements FlowEncryptor {
+    protected static final Pattern ENCRYPTED_PATTERN = Pattern.compile("^enc\\{([^\\}]+?)\\}$");
+
+    protected static final int FIRST_GROUP = 1;
+
+    protected static final String ENCRYPTED_FORMAT = "enc{%s}";
+
+    @Override
+    public abstract void processFlow(final InputStream inputStream, final OutputStream outputStream,
+                                     final PropertyEncryptor inputEncryptor, final PropertyEncryptor outputEncryptor);
```

Comment:
Since this interface method is already declared, it is not necessary to declare it again in this abstract class.